dev/ColorPicker/ColorPicker.cpp

```diff
@@ -961,18 +955,19 @@ void ColorPicker::OnHexTextChanging(winrt::TextBox const& /*sender*/, winrt::Tex
     // We'll respond to the text change if the user has entered a valid value.
     // Otherwise, we'll do nothing except mark the text box's contents as invalid.
     bool isAlphaEnabled = IsAlphaEnabled();
-    Rgb rgbValue;
-    double alphaValue = 1.0;
-
+    Rgba rgba{};
     if (isAlphaEnabled)
     {
-        HexToRgba(m_hexTextBox.Text(), &rgbValue, &alphaValue);
+        rgba = HexToRgba(m_hexTextBox.Text());
     }
```

Comment:
this helper could return a std::tuple and then unpack into locals -- see [https://en.cppreference.com/w/cpp/language/structured_binding](https://en.cppreference.com/w/cpp/language/structured_binding). If the HexToRgba helper returned a tuple then you could also write this if/else as a lambda that returns a tuple:

```c++
auto [rgbValue, alphaValue] = [this]() {
    if (isAlphaEnabled)
    {
        return HexToRgba(m_hexTextBox.Text());
    }
    else
    {
        return std::make_tuple(HexToRgb(m_hexTextBox.Text()), 1.0);
    }
}();
```